### ISSUE TITLE
is_dataclass causes a graph break

### ISSUE THREAD

**Issue by vmoens** [pytorch/pytorch#141431]
### 🐛 Describe the bug

dataclasses.is_dataclass is not inlined:
```python
from dataclasses import is_dataclass

import torch


@torch.compile(fullgraph=True)
def func(x):
    if not is_dataclass(x):
        return x + 1

func(torch.ones(()))
```

This (copy pasting the function from dataclass) runs though
```python

from dataclasses import GenericAlias, _FIELDS
def is_dataclass_recode(obj):
    """Returns True if obj is a dataclass or an instance of a
    dataclass."""
    cls = obj if isinstance(obj, type) and not isinstance(obj, GenericAlias) else type(obj)
    return hasattr(cls, _FIELDS)

import torch


@torch.compile(fullgraph=True)
def func(x):
    if not is_dataclass_recode(x):
        return x + 1

func(torch.ones(()))
```

### Error logs

```
  File "/Users/vmoens/venv/rl2/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 708, in call_function
    unimplemented(msg)
  File "/Users/vmoens/venv/rl2/lib/python3.10/site-packages/torch/_dynamo/exc.py", line 317, in unimplemented
    raise Unsupported(msg, case_name=case_name)
torch._dynamo.exc.Unsupported: 'skip function is_dataclass in file /opt/homebrew/Cellar/python@3.10/3.10.14_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/dataclasses.py'

from user code:
   File "/Users/vmoens/Library/Application Support/JetBrains/PyCharm2023.3/scratches/scratch_10.py", line 8, in func
    if not is_dataclass(x):

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True

```

### Solution
Happy to add it to `manual_torch_name_rule_map` but there are only torch objects there - is it the right place to add this sort of builtin?

### Versions

nightlies

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @amjames

**Comment by penguinwu** [comment]
cc @williamwen42 William, is this graph-break fixable?

**Comment by williamwen42** [comment]
Does https://github.com/pytorch/pytorch/pull/141294 fix it? cc @StrongerXi

**Comment by StrongerXi** [comment]
> Does #141294 fix it? cc @StrongerXi

Yep, this passes without graph break on latest main 9012e7a62f7, closed by #141294.
